rules/eq-by-simplification.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+const R = require('ramda');
+const isCalling = require('../ast-helper').isCalling;
+
+const create = context => ({
+    CallExpression(node) {
+        const match = isCalling({
+            name: 'eqBy',
+            arguments: R.both(
+                R.propSatisfies(R.lt(0), 'length'),
```

Comment:
`complement(isEmpty)` would be more readable.

Also, this is an idea for a new rule 🤔